src/utilities/globals/DEV.ts

```diff
@@ -1,29 +1,13 @@
 import global from "./global";
 import { maybe } from "./maybe";
 
-// To keep string-based find/replace minifiers from messing with __DEV__ inside
-// string literals or properties like global.__DEV__, we construct the "__DEV__"
-// string in a roundabout way that won't be altered by find/replace strategies.
-const __ = "__";
-const GLOBAL_KEY = [__, __].join("DEV");
-
 function getDEV() {
-  try {
-    return Boolean(__DEV__);
-  } catch {
-    Object.defineProperty(global, GLOBAL_KEY, {
-      // In a buildless browser environment, maybe(() => process.env.NODE_ENV)
-      // evaluates as undefined, so __DEV__ becomes true by default, but can be
-      // initialized to false instead by a script/module that runs earlier.
-      value: maybe(() => process.env.NODE_ENV) !== "production",
-      enumerable: false,
-      configurable: true,
-      writable: true,
-    });
-    // Using computed property access rather than global.__DEV__ here prevents
-    // string-based find/replace strategies from munging this to global.false:
-    return (global as any)[GLOBAL_KEY];
-  }
+  return "__DEV__" in global
+    ? Boolean(global.__DEV__)
+    // In a buildless browser environment, maybe(() => process.env.NODE_ENV)
+    // evaluates as undefined, so __DEV__ becomes true by default, but can be
+    // initialized to false instead by a script/module that runs earlier.
+    : maybe(() => process.env.NODE_ENV) !== "production";
```

Comment:
If we use `typeof`, we could get around `global` and the `maybe` call. (`typeof` will not cause a `ReferenceError` when referencing a non-existing variable.)

There might be some reason for `global` and accesing global scopes in a specific order though, so please correct me on that if I'm wrong.

```suggestion
  return typeof __DEV__ !== "undefined" ? Boolean(__DEV__)
    : typeof process !== "undefined" && process?.env?.NODE_ENV !== "production";
```

There could also be some merit by exporting `getDev` instead of the result of executing it, in cases where `__DEV__` is changed during runtime or [AsyncLocalStorage](https://nodejs.org/api/async_context.html) is used to keep multiple different global contexts.

Reply:
Thinking about this more, and going completely in the other direction: I believe this could tree-shake a *lot* better if we directly calculated it on top-level instead of a function call. Would that be an option?

Reply:
We want an existing global definition of `__DEV__` (such as in React Native, or from manual configuration) to take precedence, so checking `global.__DEV__` is still useful for that.

Reply:
Also, please do not replace code like `maybe(() => process.env.NODE_ENV)` with `process?.env?.NODE_ENV`, because that compiles to JavaScript that minifiers will have trouble replacing with a constant.

The value of `maybe` is that it allows you to avoid the optional chaining syntax, gets the return type right, and still allows compilers to replace the original `process.env.NODE_ENV` expression (because it remains intact in the `maybe(() => ...)` function body).